dds/DCPS/TypeSupportImpl.cpp

```diff
@@ -49,80 +51,125 @@ TypeSupportImpl::get_type_name()
   return type._retn();
 }
 
-void TypeSupportImpl::to_type_info(XTypes::TypeInformation& type_info) const
+void TypeSupportImpl::to_type_info_i(XTypes::TypeIdentifierWithDependencies& ti_with_deps,
+                                     const XTypes::TypeIdentifier& ti,
+                                     const XTypes::TypeMap& type_map) const
 {
-  using namespace XTypes;
-  const TypeIdentifier& minTypeId = getMinimalTypeIdentifier();
-  const TypeMap& minTypeMap = getMinimalTypeMap();
-  const TypeMap::const_iterator pos = minTypeMap.find(minTypeId);
-
-  if (pos == minTypeMap.end()) {
-    type_info.minimal.typeid_with_size.type_id = TypeIdentifier();
-    type_info.minimal.typeid_with_size.typeobject_serialized_size = 0;
-
+  const XTypes::TypeMap::const_iterator pos = type_map.find(ti);
+
+  if (pos == type_map.end()) {
+    ACE_ERROR((LM_ERROR, ACE_TEXT("(%P|%t) ERROR: TypeSupportImpl::to_type_info_i, ")
+               ACE_TEXT("%C TypeIdentifier %C of topic type not found in local type map.\n"),
+               //kind_to_string(ti.kind()), XTypes::equivalence_hash_to_string(ti.equivalence_hash()).c_str()));
```

Comment:
kind_to_string is commented out here and a few other places because it causes problems building test libraries.  It needs to be resolved and uncommented.